File: packages/snow_draw_core/lib/draw/types/edit_operation_id.dart
/// Stable identifier for an edit-domain operation (move/resize/rotate/...).
typedef EditOperationId = String;

/// Built-in edit operation ids used by the edit session and operation registry.
final class EditOperationIds {
  EditOperationIds._();

  static const move = 'move';
  static const resize = 'resize';
  static const rotate = 'rotate';
  static const freeTransform = 'free_transform';
  static const arrowPoint = 'arrow_point';
}

File: packages/snow_draw_rust_engine/src/draw/types/edit_operation_id.rs
#![allow(dead_code)]

/// Stable identifier for an edit-domain operation.
pub type EditOperationId = &'static str;

/// Built-in edit operation identifiers.
pub struct EditOperationIds;

impl EditOperationIds {
    pub const MOVE: EditOperationId = "move";
    pub const RESIZE: EditOperationId = "resize";
    pub const ROTATE: EditOperationId = "rotate";
    pub const CONNECTOR_POINT: EditOperationId = "connector_point";
    pub const ARROW_POINT: EditOperationId = Self::CONNECTOR_POINT;
}
